src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -96,6 +99,31 @@ public AppiumServiceBuilder() {
         withEnvironment(System.getenv());
     }
 
+    /**
+     * Provides a measure of how strongly this {@link DriverService} supports the given
+     * {@code capabilities}. A score of 0 or less indicates that this {@link DriverService} does not
+     * support instances of {@link org.openqa.selenium.WebDriver} that require {@code capabilities}.
+     * Typically, the score is generated by summing the number of capabilities that the driver
+     * service directly supports that are unique to the driver service (that is, things like
+     * "{@code proxy}" don't tend to count to the score).
+     * Higher the score, higher the possibility of getting grid sessions created sooner.
+     */
+    @Override public int score(Capabilities capabilites) {
+        int score = 0;
+
+        if (capabilites.getCapability(PLATFORM_NAME) != null) {
+            score++;
+        }
+
+        String browserName = capabilites.getBrowserName();
```

Comment:
may this ever return null?

Reply:
ah, hm. It cannot : `return String.valueOf(Optional.ofNullable(getCapability("browserName")).orElse(""));`